### DISCUSSION TITLE
Where to find IDs for different object such as robot, gripper, camera, sponge, etc.

### DISCUSSION THREAD

**Discussion by architjain19** [empriselab/RCareWorld#105] [Help - Programming] [answered]
Where can i find the get the ID of an object such as 
- robot
- camera
- gripper
- sponge/scrubber

what is to be provided as `id` in `env.GetAttr(id)` because the examples codes doesn't provides ID of all objects.

**Comment by YoruCathy** [comment, marked as answer]
Hi @architjain19 check the output of print(env.attrs)
`{98721: <pyrcareworld.attributes.camera_attr.CameraAttr object at 0x7fd4f5f945b0>, 4857: <pyrcareworld.attributes.sponge_score_attr.SpongeScoreAttr object at 0x7fd4f5f94670>, 2333: <pyrcareworld.attributes.base_attr.BaseAttr object at 0x7fd4f5f94760>, 91846: <pyrcareworld.attributes.sponge_attr.SpongeAttr object at 0x7fd4f5f947c0>, 654321: <pyrcareworld.attributes.camera_attr.CameraAttr object at 0x7fd4f5f94820>, 2215820: <pyrcareworld.attributes.controller_attr.ControllerAttr object at 0x7fd4f5f94880>, 221582: <pyrcareworld.attributes.controller_attr.ControllerAttr object at 0x7fd4f6146230>, 758666: <pyrcareworld.attributes.base_attr.BaseAttr object at 0x7fd4f6145db0>}`

Now you have information about all the objects in the scene. For each object,
```
myobj = env.GetAttr(2333)
print(myobj.data)
```
Now you have the data of this object. For instance, the object with id=2333 is the red cube. You can try changing the position of it, or move the end effector of the robot to that specific position. 

Check the example test_bathing.py to get an overall idea about how to use the camera, robot, gripper, and sponge.

**Comment by architjain19** [comment]
Well i got this, thanks.
But i got a doubt on how will the position and other data will update in realtime as robot moves in the simulation. Since we are creating an instance of an object once, the robot position is stored only once during the env initialization. Where do the robot pose gets update, even tried create new objects after time t in simulation, yet the position is same as initial one.

**Reply by YoruCathy** [reply]
Try robot.positions[0]
